crates/aptos/src/update/movefmt.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{get_additional_binaries_dir, update_binary, BinaryUpdater, UpdateRequiredInfo};
+use crate::common::{
+    types::{CliCommand, CliTypedResult},
+    utils::cli_build_information,
+};
+use anyhow::{anyhow, bail, Context, Result};
+use aptos_build_info::BUILD_OS;
+use async_trait::async_trait;
+use clap::Parser;
+use self_update::{backends::github::Update, update::ReleaseUpdate};
+use std::path::PathBuf;
+
+const FORMATTER_BINARY_NAME: &str = "movefmt";
+const TARGET_FORMATTER_VERSION: &str = "1.0.0.beta";
+
+const FORMATTER_EXE_ENV: &str = "FORMATTER_EXE";
+#[cfg(target_os = "windows")]
+const FORMATTER_EXE: &str = "movefmt.exe";
+#[cfg(not(target_os = "windows"))]
+const FORMATTER_EXE: &str = "movefmt";
+
+/// Update Movefmt, the tool used for formatting Move code.
+#[derive(Debug, Parser)]
+pub struct FormatterUpdateTool {
```

Comment:
There is lots of partial duplication throughout this file now with the revela code, can you try abstract it out?

Reply:
I extracted some of the common code into a new file `update_helper.rs`.